File: findTheNumberOfWaysToPlacePeople.py
from typing import List

class Solution:
    def numberOfPairs(self, points: List[List[int]]) -> int:
        n = len(points)
        ans = 0
        for i in range(n):
            ax, ay = points[i]
            for j in range(n):
                if i == j:
                    continue
                bx, by = points[j]
                # A must be upper-left of B (lines allowed)
                if ax <= bx and ay >= by and (ax != bx or ay != by):
                    # Check if rectangle (including border) is empty
                    blocked = False
                    minx, maxx = ax, bx
                    miny, maxy = by, ay
                    for k in range(n):
                        if k == i or k == j:
                            continue
                        xk, yk = points[k]
                        if minx <= xk <= maxx and miny <= yk <= maxy:
                            blocked = True
                            break
                    if not blocked:
                        ans += 1
        return ans

File: findTheNumberOfWaysToPlacePeople.go
package main

func numberOfPairs(points [][]int) int {
	n := len(points)
	ans := 0
	for i := 0; i < n; i++ {
		ax, ay := points[i][0], points[i][1]
		for j := 0; j < n; j++ {
			if i == j {
				continue
			}
			bx, by := points[j][0], points[j][1]
			// A is upper-left of B (lines allowed)
			if ax <= bx && ay >= by && (ax != bx || ay != by) {
				blocked := false
				minx, maxx := ax, bx
				miny, maxy := by, ay
				for k := 0; k < n; k++ {
					if k == i || k == j {
						continue
					}
					xk, yk := points[k][0], points[k][1]
					if xk >= minx && xk <= maxx && yk >= miny && yk <= maxy {
						blocked = true
						break
					}
				}
				if !blocked {
					ans++
				}
			}
		}
	}
	return ans
}
